src/ledger/LedgerTxn.cpp

```diff
@@ -1749,22 +1948,64 @@ LedgerTxnRoot::Impl::getNewestVersion(LedgerKey const& key) const
         return getFromEntryCache(cacheKey);
     }
 
+    auto prefetched = mPrefetcher->getPrefetched(key);
+    if (prefetched.first)
+    {
+        return prefetched.second;
+    }
+
+    auto keysToPrefetch = mPrefetcher->getBatchToPrefetch(key.type());
+    bool doPrefetch = !keysToPrefetch.empty();
+    keysToPrefetch.insert(key);
+
     std::shared_ptr<LedgerEntry const> entry;
     try
     {
         switch (key.type())
         {
         case ACCOUNT:
-            entry = loadAccount(key);
+            if (!doPrefetch)
```

Comment:
I would prefer to split that method into several smaller ones:
- if (doPrefetch) condition could be make top one
- loadEntry and bulkLoadEntries methods could be extracted, first one would do what that try block was doing before the PR, and bulkLoadEntries only do the prefetching branch:


```
if (doPrefetch)
{
  bulkLoadEntries(keysToPrefetch);
  entry = updatePrefetcherCache(res, key);
}
else
{
  entry = loadAccount(key);
}
```